app/services/operating_system_parser.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+# Methods to parse facts related to the OS
+module OperatingSystemParser
+  def operatingsystem
+    args = { :name => os_name, :major => os_major, :minor => os_minor }
+    args[:release_name] = os_release_name if os_name == 'Debian' || os_name == 'Ubuntu'
+    return @local_os if local_os(args).present?
+    return @new_os if new_os(args).present?
+    logger.debug do
+      'Ansible facts parser: No OS could be created with '\
+      "os_name='#{os_name}' os_major='#{os_major}' "\
+      "os_minor='#{os_minor}': "\
+      "#{@new_os.errors if @new_os.present?}"
+    end
+    nil
+  end
+
+  def local_os(args)
+    @local_os = Operatingsystem.where(args).first
+  end
+
+  def new_os(args)
+    return @new_os if @new_os.present?
+    @new_os = Operatingsystem.new(args.merge(:description => os_description))
+    @new_os if @new_os.valid? && @new_os.save
+  end
+
+  def debian_os_major_sid
+    case facts[:ansible_distribution_major_version]
```

Comment:
This feels like a bad design. It's strictly tied to Ansible, but nothing in the name of the class suggests that. If anything, It should be named `AnsibleOperatingSystemParser`.

Reply:
yeah, originally this code lived in `ForemanAnsible` namespace so it made sense, but as part of the move, the class should be either renamed or kept under `ForemanAnsible` module. I don't mind much which way, since after this is in, we should start deduplicating the logic and my hope is, as part of that effort, this class will be dropped entirely.